### PR TITLE
Add docs for the global config file, registry, manifest changes and publish

### PR DESCRIPTION
**Pull request by minhqdao**
Add docs for https://github.com/fortran-lang/fpm/pull/817 and https://github.com/fortran-lang/fpm/pull/876:

- [x] How to use a global configuration file.
- [x] How to setup a local registry.
- [x] How to use a remote registry.
- [x] Additional manifest options (`namespace` and `v`).
- [x] How to upload a package to the official registry using `fpm publish`.

### CODE REVIEW

**Review by arteevraina — APPROVED** [review]
I would still considering adding a note warning regarding this first alpha version of registry.

**Comment by minhqdao** [comment]
> I would still considering adding a note warning regarding this first alpha version of registry.
> Overall the PR looks good to me.

Good point, it's been added.

**Comment by minhqdao** [comment]
I'd appreciate if at least one native English speaker could review this. @everythingfunctional?

**Review by henilp105 — APPROVED** [review]
Thanks @minhqdao , looks good to me.

File: pages/spec/manifest.md [hunk 1]
```diff
@@ -530,6 +573,28 @@ For more verbose layout use normal tables rather than inline tables to specify d
...
+
+#### Version
+
+If you want to download a specific version of a package instead of the newest one available, you can specify the version (`v`) in the manifest.
+
+```toml
+[dependencies]
+example-package.namespace = "example-namespace"
+example-package.v = "1.0.0"
+```
```
Comment (everythingfunctional) [inline pages/spec/manifest.md:597]: This is kind of an interesting design. Not exactly what I would have initially thought of, but I suppose it works.

File: pages/spec/publish.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+A package must specify a valid [SPDX](https://spdx.org/licenses/) license in its manifest.
+
+### Upload rights
+
+A package can only be uploaded by package admins and maintainers. Package admins can grant admin and maintainer rights to other users.
+
+## Uploading
+
+fpm will create a tarball and upload the package to the registry when all then requirements are met. You can see all the data being sent to the registry without uploading by running `fpm publish --show-form-data`.
```
Comment (everythingfunctional) [inline pages/spec/publish.md:37]:
```suggestion
fpm will create a tarball and upload the package to the registry when all the requirements are met. You can see all the data being sent to the registry without uploading by running `fpm publish --show-form-data`.
```

Typo

**Review by everythingfunctional — APPROVED** [review]
Everything else looks good. Great work everybody.

**Comment by minhqdao** [comment]
Thank you for your reviews @everythingfunctional @arteevraina @henilp105

**Pull request merged**